### DISCUSSION TITLE
meepctl dockerize

### DISCUSSION THREAD

**Discussion by itnor4** [InterDigitalInc/AdvantEDGE#352] [Q&A]
Hi All,

I am trying to install the new AdvantEDGE version, for working on the Demo3. I have used different version of nvm i.e. 12.19.0 and npm version 8.5.3. Other than that I followed all the steps, the meepctl dockerize all is not working. I tried to dockerize each image, but the meepctl-auth-svc doesn't work. Please find the attached log for you're reference and let me know if more details are needed. Thank you.

ubuntu@advantedge18:~/AdvantEDGE/go-apps/meepctl$ meepctl build all --nolint
Using repo config file: /home/ubuntu/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/ubuntu/.meepctl.yaml
-- meep-auth-svc --
   + checking external dependencies
   + building meep-auth-svc

-- meep-mon-engine --
   + checking external dependencies
   + building meep-mon-engine

-- meep-platform-ctrl --
   + checking external dependencies
   + building meep-platform-ctrl

-- meep-virt-engine --
   + checking external dependencies
   + building meep-virt-engine

-- meep-webhook --
   + checking external dependencies
   + building meep-webhook

-- meep-frontend --
   + checking external dependencies
   + checking local dependencies
     * meep-sandbox-ctrl-api
     * meep-mon-engine-api
     * meep-gis-engine-api
     * meep-auth-svc-client
     * meep-platform-ctrl-api
   + building meep-frontend

-- meep-ams --
   + checking external dependencies
   + building meep-ams

-- meep-app-enablement --
   + checking external dependencies
   + building meep-app-enablement

-- meep-gis-engine --
   + checking external dependencies
   + building meep-gis-engine

-- meep-loc-serv --
   + checking external dependencies
   + building meep-loc-serv

-- meep-metrics-engine --
   + checking external dependencies
   + building meep-metrics-engine

-- meep-mg-manager --
   + checking external dependencies
   + building meep-mg-manager

-- meep-rnis --
   + checking external dependencies
   + building meep-rnis

-- meep-sandbox-ctrl --
   + checking external dependencies
   + building meep-sandbox-ctrl

-- meep-tc-engine --
   + checking external dependencies
   + building meep-tc-engine

-- meep-tc-sidecar --
   + checking external dependencies
   + building meep-tc-sidecar

-- meep-wais --
   + checking external dependencies
   + building meep-wais

ubuntu@advantedge18:~/AdvantEDGE/go-apps/meepctl$ meepctl deploy dep
Using repo config file: /home/ubuntu/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/ubuntu/.meepctl.yaml
Using docker registry: meep-docker-registry:30001
Deployed meep-couchdb
Deployed meep-docker-registry
Deployed meep-grafana
Deployed meep-influxdb
Deployed meep-ingress
Deployed meep-kube-state-metrics
Deployed meep-open-map-tiles
Deployed meep-postgis
Deployed meep-prometheus
Deployed meep-redis
ubuntu@advantedge18:~/AdvantEDGE/go-apps/meepctl$ 
ubuntu@advantedge18:~/AdvantEDGE/go-apps/meepctl$ meepctl dockerize all
Using repo config file: /home/ubuntu/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/ubuntu/.meepctl.yaml
Using docker registry: meep-docker-registry:30001
-- meep-auth-svc --
   + copy docker file
   + copy api files
   + copy docker data
   + dockerize meep-auth-svc

**Comment by roymx** [comment]
Hello @itnor4 ,

please, provide the output of the following command?
`meepctl dockerize all -v`

**Reply by itnor4** [reply]
Hi @roymx,

Thank you for the reply. Output of the following command

ubuntu@advantedge18:~/AdvantEDGE/go-apps/meepctl$ meepctl dockerize all -v
Using repo config file: /home/ubuntu/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/ubuntu/.meepctl.yaml
Dockerize called
[arg]  targets: [all]
[flag] registry: 
[flag] verbose: true
[flag] time: false
Using docker registry: meep-docker-registry:30001
-- meep-auth-svc --
   + copy docker file
Cmd: [cp Dockerfile /home/ubuntu/AdvantEDGE/bin/meep-auth-svc]
Result: OK
   + copy api files
Cmd: [rm -r /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api]
Result: OK
Cmd: [mkdir -p /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api]
Result: OK
    Using: /home/ubuntu/AdvantEDGE/go-apps/meep-auth-svc/api/swagger.yaml --> /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API
Cmd: [rm /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API]
Result: rm: cannot remove '/home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API': No such file or directory

Cmd: [cp /home/ubuntu/AdvantEDGE/go-apps/meep-auth-svc/api/swagger.yaml /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API]
Result: OK
Cmd: [rm -r /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/user-api]
Result: OK
Cmd: [mkdir -p /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/user-api]
Result: OK
   + copy docker data
Cmd: [rm -r /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data]
Result: OK
Cmd: [mkdir -p /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data]
Result: OK
    Using: /home/ubuntu/AdvantEDGE/config/permissions-open.yaml --> /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml
Cmd: [rm -r /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml]
Result: rm: cannot remove '/home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml': No such file or directory

Cmd: [cp -r /home/ubuntu/AdvantEDGE/config/permissions-open.yaml /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml]
Result: OK
    Using: /home/ubuntu/AdvantEDGE/go-apps/meep-auth-svc/entrypoint.sh --> /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh
Cmd: [rm -r /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh]
Result: rm: cannot remove '/home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh': No such file or directory

Cmd: [cp -r /home/ubuntu/AdvantEDGE/go-apps/meep-auth-svc/entrypoint.sh /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh]
Result: OK
Cmd: [/bin/sh -c find /home/ubuntu/AdvantEDGE/bin/meep-auth-svc -type f | xargs sha256sum | sort | sha256sum]
Result: sha256sum: /home/ubuntu/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE: No such file or directory
sha256sum: Auth: No such file or directory
sha256sum: Service: No such file or directory
sha256sum: REST: No such file or directory
sha256sum: API: No such file or directory
eb253d63166860c007880a526f40b7a1b5afc9fb37081573710df7b0e16ee123  -

   + dockerize meep-auth-svc
Cmd: [docker build --no-cache --rm --label MeepVersion=sha256sum: -t meep-docker-registry:30001/meep-auth-svc /home/ubuntu/AdvantEDGE/bin/meep-auth-svc]

**Reply by roymx** [reply]
Hello @itnor4 ,
It looks like some logs are missing, maybe the output was not fully copied or was copied before the command had completed.

I attached the output that I get over here; Please re-post with the outcome of `Cmd: [docker build --no-cache --rm --label MeepVersion=sha256sum: -t meep-docker-registry:30001/meep-auth-svc /home/ubuntu/AdvantEDGE/bin/meep-auth-svc]`

Thanks!


```
$ meepctl dockerize meep-auth-svc -v
Using repo config file: /home/<some-user>/AdvantEDGE/.meepctl-repocfg.yaml
Using meepctl config file: /home/<some-user>/.meepctl.yaml
Dockerize called
[arg]  targets: [meep-auth-svc]
[flag] registry:
[flag] verbose: true
[flag] time: false
Using docker registry: meep-docker-registry:30001
-- meep-auth-svc --
   + copy docker file
Cmd: [cp Dockerfile /home/<some-user>/AdvantEDGE/bin/meep-auth-svc]
Result: OK
   + copy api files
Cmd: [rm -r /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api]
Result: OK
Cmd: [mkdir -p /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api]
Result: OK
    Using: /home/<some-user>/AdvantEDGE/go-apps/meep-auth-svc/api/swagger.yaml --> /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API
Cmd: [rm /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API]
Result: rm: cannot remove '/home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API': No such file or directory

Cmd: [cp /home/<some-user>/AdvantEDGE/go-apps/meep-auth-svc/api/swagger.yaml /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE Auth Service REST API]
Result: OK
Cmd: [rm -r /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/user-api]
Result: OK
Cmd: [mkdir -p /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/user-api]
Result: OK
   + copy docker data
Cmd: [rm -r /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data]
Result: OK
Cmd: [mkdir -p /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data]
Result: OK
    Using: /home/<some-user>/AdvantEDGE/go-apps/meep-auth-svc/entrypoint.sh --> /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh
Cmd: [rm -r /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh]
Result: rm: cannot remove '/home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh': No such file or directory

Cmd: [cp -r /home/<some-user>/AdvantEDGE/go-apps/meep-auth-svc/entrypoint.sh /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/entrypoint.sh]
Result: OK
    Using: /home/<some-user>/AdvantEDGE/config/permissions-open.yaml --> /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml
Cmd: [rm -r /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml]
Result: rm: cannot remove '/home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml': No such file or directory

Cmd: [cp -r /home/<some-user>/AdvantEDGE/config/permissions-open.yaml /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/data/permissions.yaml]
Result: OK
Cmd: [/bin/sh -c find /home/<some-user>/AdvantEDGE/bin/meep-auth-svc -type f | xargs sha256sum | sort | sha256sum]
Result: sha256sum: /home/<some-user>/AdvantEDGE/bin/meep-auth-svc/api/AdvantEDGE: No such file or directory
sha256sum: Auth: No such file or directory
sha256sum: Service: No such file or directory
sha256sum: REST: No such file or directory
sha256sum: API: No such file or directory
fb30f36fbbb1a98bfd950e6161f5cb237b0c030a4af93ee01102aedff976a187  -

   + dockerize meep-auth-svc
Cmd: [docker build --no-cache --rm --label MeepVersion=sha256sum: -t meep-docker-registry:30001/meep-auth-svc /home/<some-user>/AdvantEDGE/bin/meep-auth-svc]
Result: Sending build context to Docker daemon  21.95MB
Step 1/10 : FROM debian:9.6-slim
 ---> 3a08df3c763e
Step 2/10 : COPY ./meep-auth-svc /meep-auth-svc
 ---> 4038ef3c79d0
Step 3/10 : COPY ./api /api
 ---> d495785a6f5c
Step 4/10 : COPY ./user-api /user-api
 ---> 6f0eb832ac8c
Step 5/10 : COPY ./data /
 ---> 8d54140382f3
Step 6/10 : RUN chmod +x /entrypoint.sh
 ---> Running in 6f18004d78d1
Removing intermediate container 6f18004d78d1
 ---> bfc9c0d62b3e
Step 7/10 : RUN apt-get update  && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends ca-certificates
 ---> Running in 5504e53a675d
Get:1 http://security.debian.org/debian-security stretch/updates InRelease [53.0 kB]
Ign:2 http://deb.debian.org/debian stretch InRelease
Get:3 http://deb.debian.org/debian stretch-updates InRelease [93.6 kB]
Get:4 http://deb.debian.org/debian stretch Release [118 kB]
Get:5 http://deb.debian.org/debian stretch Release.gpg [3177 B]
Get:6 http://security.debian.org/debian-security stretch/updates/main amd64 Packages [758 kB]
Get:7 http://deb.debian.org/debian stretch/main amd64 Packages [7080 kB]
Fetched 8106 kB in 3s (2511 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libssl1.1 openssl
The following NEW packages will be installed:
  ca-certificates libssl1.1 openssl
0 upgraded, 3 newly installed, 0 to remove and 21 not upgraded.
Need to get 2278 kB of archives.
After this operation, 5408 kB of additional disk space will be used.
Get:1 http://security.debian.org/debian-security stretch/updates/main amd64 libssl1.1 amd64 1.1.0l-1~deb9u5 [1361 kB]
Get:2 http://security.debian.org/debian-security stretch/updates/main amd64 openssl amd64 1.1.0l-1~deb9u5 [749 kB]
Get:3 http://security.debian.org/debian-security stretch/updates/main amd64 ca-certificates all 20200601~deb9u2 [168 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 2278 kB in 0s (3139 kB/s)
Selecting previously unselected package libssl1.1:amd64.
(Reading database ... 6318 files and directories currently installed.)
Preparing to unpack .../libssl1.1_1.1.0l-1~deb9u5_amd64.deb ...
Unpacking libssl1.1:amd64 (1.1.0l-1~deb9u5) ...
Selecting previously unselected package openssl.
Preparing to unpack .../openssl_1.1.0l-1~deb9u5_amd64.deb ...
Unpacking openssl (1.1.0l-1~deb9u5) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20200601~deb9u2_all.deb ...
Unpacking ca-certificates (20200601~deb9u2) ...
Processing triggers for libc-bin (2.24-11+deb9u3) ...
Setting up libssl1.1:amd64 (1.1.0l-1~deb9u5) ...
Setting up openssl (1.1.0l-1~deb9u5) ...
Setting up ca-certificates (20200601~deb9u2) ...
Updating certificates in /etc/ssl/certs...
137 added, 0 removed; done.
Processing triggers for libc-bin (2.24-11+deb9u3) ...
Processing triggers for ca-certificates (20200601~deb9u2) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Removing intermediate container 5504e53a675d
 ---> 64bc5a012b88
Step 8/10 : RUN update-ca-certificates
 ---> Running in 109662b67951
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Removing intermediate container 109662b67951
 ---> 76c50718788d
Step 9/10 : ENTRYPOINT ["/entrypoint.sh"]
 ---> Running in ef16fc79e0f6
Removing intermediate container ef16fc79e0f6
 ---> b23848d971f0
Step 10/10 : LABEL MeepVersion=sha256sum:
 ---> Running in 547796c5dd5d
Removing intermediate container 547796c5dd5d
 ---> cf252e37f619
Successfully built cf252e37f619
Successfully tagged meep-docker-registry:30001/meep-auth-svc:latest

Cmd: [docker push meep-docker-registry:30001/meep-auth-svc]
Result: The push refers to repository [meep-docker-registry:30001/meep-auth-svc]
a66cfc812b77: Preparing
fc2dcc120d9a: Preparing
08097dff58a2: Preparing
7ac7e338a3d3: Preparing
31371e19decd: Preparing
33b30c3efb29: Preparing
bb447b910018: Preparing
3c816b4ead84: Preparing
33b30c3efb29: Waiting
3c816b4ead84: Waiting
bb447b910018: Waiting
7ac7e338a3d3: Pushed
31371e19decd: Pushed
08097dff58a2: Pushed
a66cfc812b77: Pushed
bb447b910018: Layer already exists
3c816b4ead84: Layer already exists
33b30c3efb29: Pushed
fc2dcc120d9a: Pushed
latest: digest: sha256:011b015ef401dd3fa3c7c3385971fa6bb5d41abbac23f8d712da29aed445751b size: 1993


```
